packages/enketo-core/src/js/form-model.js

```diff
@@ -1554,6 +1557,71 @@ FormModel.prototype.evaluate = function (
     }
 };
 
+/**
+ * This function checks the existence of a template node and
+ * when non existent, creates an empty node.
+ * This fixes some situations where, when loading editing data,
+ * repeat nodes aren't properly returned, crashing editing.
+ *
+ * @param {XMLDocument} recordDoc
+ */
+FormModel.prototype.fixRecordRepeatData = function (recordDoc) {
+    // Need to use the raw model without transformations
+    const modelDoc = new DOMParser().parseFromString(
+        this.data.modelStr,
+        'text/xml'
+    );
+
+    const modelInstance = modelDoc.querySelector('model > instance');
+
+    // Check for a missing template node and if so
+    // creates an empty node. This fixes some situations
+    // in editing where accessing nodes data crashes editing
+    // (not the best practice to create a function inside of another
+    // function, but it's a recursive and very contextual function only
+    // used in this function)
+    const fixNode = (node, modelNode) => {
+        // get the model element for the current instance node
+        const modelEl = Array.from(modelNode.children).find(
+            (child) => child.nodeName === node.nodeName
+        );
+
+        // If no model element is found, there's nothing to check
+        if (!modelEl) return;
+
+        const isTemplate =
+            modelEl.hasAttribute('jr:template') ||
+            modelEl.hasAttribute('template');
```

Comment:
nit: please extract a `isTemplate` function, this logic repeats in two places where such function call could be nicely inlined.